rust/libceed/src/vector.rs

```diff
@@ -375,6 +398,76 @@ impl<'a> Vector<'a> {
         self.check_error(ierr)
     }
 
+    /// Set Vector array to borrowed slice of the same length
+    ///
+    /// # arguments
+    ///
+    /// * `slice` - values to into self; length must match
+    ///
+    /// ```
+    /// # use libceed::prelude::*;
+    /// # fn main() -> libceed::Result<()> {
+    /// # let ceed = libceed::Ceed::default_init();
+    /// let mut vec = ceed.vector(4)?;
+    /// let mut slice = vec![10., 11., 12., 13.];
+    ///
+    /// {
+    ///     // `borrow` holds a mutable reference to the slice
+    ///     let borrow = vec.set_borrowed_slice(&mut slice)?;
+    ///     vec.view()?.iter().enumerate().for_each(|(i, v)| {
+    ///         assert_eq!(*v, 10. + i as Scalar, "Slice not set correctly");
+    ///     });
+    ///
+    ///     // This line will not compile, as the `borrow` is still in scope
+    ///     // slice[0] = 5.0;
+    ///
+    ///     // Changes here are reflected in the `slice`
+    ///     vec.set_value(5.0)?;
+    ///     vec.view()?.iter().for_each(|v| {
+    ///         assert_eq!(*v, 5 as Scalar, "Slice not set correctly");
+    ///     });
+    /// }
+    ///
+    /// // Slice remains changed
+    /// slice.iter().for_each(|v| {
+    ///     assert_eq!(*v, 5 as Scalar, "Slice not set correctly");
+    /// });
+    ///
+    /// // This line will error, as the Vector has dropped the reference
+    /// //   to the slice and no new array was allocated for the Vector
+    /// // vec.view()?;
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub fn set_borrowed_slice<'b>(
+        &mut self,
+        slice: &'b mut [crate::Scalar],
+    ) -> crate::Result<VectorSliceBorrow<'b>> {
```

Comment:
Sorry to lose track of this. My concern is with naming. Seems to me a slice is always borrowed and this terminology (but not type signature) is ambiguous as to whether the slice is provided to the vector or the vector is setting the slice. But what about `wrap_slice` or perhaps `wrap_slice_mut`, which "Uses pre-sized `Vector` to wrap a provided mutable slice". Looks useful and clean otherwise.

Reply:
Good point. I switched over to `wrapper` based language.